lib/phoenix_library/books.ex

```diff
@@ -0,0 +1,202 @@
+defmodule PhoenixLibrary.Books do
+  @moduledoc """
+  Fornece funções para manipular registros de Book no banco de dados:
+
+  - Função `create_book/1` para criar um novo livro.
+  - Função `load_book/1` para carregar um livro existente.
+  - Função `update_book/1` para atualizar um livro existente.
+  - Função `delete_book/1` para deletar um livro existente.
+  - Função `list_books/0` para listar todos os livros existentes no banco de dados.
+  - Função `by_title/1` para pesquisar livros por título.
+  - Função `by_authorship/1` para pesquisar livros por autoria.
+  """
+
+  import Ecto.Query, only: [from: 2]
+
+  alias PhoenixLibrary.{Books.Book, Repo}
+
+  @doc """
+  Retorna uma tupla {:ok, <livro criado>} caso o livro seja criado com sucesso ou
+  uma tupla {:erro, <changeset>} caso o livro não possa ser criado.
+
+  ## Parâmetros
+
+    - params: Map com dados para criar um novo livro no banco de dados.
+
+  ## Exemplos
+
+    iex> PhoenixLibrary.Books.create_book(%{
+           title: "Adopting Elixir",
+           publisher: "The Pragmatic Programmers",
+           authorship: ["Ben Marx", "José Valim", "Bruce Tate"],
+           cover_photo: "https://pragprog.com/titles/tvmelixir/adopting-elixir/tvmelixir.jpg"
+         })
+
+    {:ok, %PhoenixLibrary.Books.Book{...}}
+
+    iex> PhoenixLibrary.Books.create_book(%{
+           title: "Adopting Elixir"
+         })
+
+    {:error, #Ecto.Changeset<..., valid?: false>}
+  """
+  def create_book(params) do
```

Comment:
O que acha de adicionar specs nas funções públicas? Não chequei se você tá usando dialyzer, mas isso ajuda tanto a entender o que esperar de retorno da função quanto a forçar que o resultado seja sempre o esperado
```suggestion
  @spec create_book(map()) :: {:ok, Book.t()} | {:error, Ecto.Changeset.t()}
  def create_book(params) do
```